features/dd-sdk-android-trace/src/main/java/com/datadog/trace/bootstrap/instrumentation/api/AgentTracer.java

```diff
@@ -233,6 +235,7 @@ AgentSpan startSpan(
 
     AgentScope activateNext(AgentSpan span);
 
+    @Nullable
```

Comment:
> [!NOTE]
> That is a nice idea, but shouldn't we add `@Nullable`/`@NonNull` to all methods' return value and arguments, just to be safe?

Reply:
This is a huge task and should not be part of this epic, for now I am adding this whenever I interact with the Java code from Kotlin because this is where the problem can come from. As a rule of thumb we should always check the Java code when calling from Kotlin to make sure we are safe.